packages/eas-cli/src/utils/expoCli.ts

```diff
@@ -1,8 +1,60 @@
+import { getConfig } from '@expo/config';
 import spawnAsync from '@expo/spawn-async';
 import chalk from 'chalk';
+import { boolish } from 'getenv';
 import resolveFrom from 'resolve-from';
+import semver from 'semver';
 
 import Log from '../log';
+import { memoize } from './expodash/memoize';
+
+// Aggressively returns `true` (UNVERSIONED, invalid SDK version format) to push users towards the versioned CLI.
+function gteSdkVersion(fromSdkVersion: string, sdkVersion: string): boolean {
+  if (fromSdkVersion === 'UNVERSIONED') {
+    return true;
+  }
+
+  try {
+    return semver.gte(fromSdkVersion, sdkVersion);
+  } catch {
+    return true;
+  }
+}
+
+/**
+ * @returns `true` if the project is SDK +46, has `@expo/cli`, and `EXPO_USE_LOCAL_CLI` is not set to a _false_ value.
+ */
+export function shouldUseVersionedExpoCLIExpensive(projectDir: string): boolean {
+  // Users can disable local CLI settings EXPO_USE_LOCAL_CLI=false
+  // https://github.com/expo/expo/blob/69eddda7bb1dbfab44258f468cf7f22984c1e44e/packages/expo/bin/cli.js#L10
+  // Run the environment variable check first as it's the cheapest.
+  const userDefinedVersionedCliEnabled = boolish('EXPO_USE_LOCAL_CLI', true);
+  if (!userDefinedVersionedCliEnabled) {
+    return false;
+  }
+
+  try {
+    // NOTE(EvanBacon): The CLI package could be available through a monorepo
+    // we need to ensure the project is specifically using a known supported Expo SDK version.
+    const { sdkVersion } = getConfig(projectDir).exp;
```

Comment:
Don't call `getConfig` explicitly. This should be passed from the command's context. Talk with @wschurman if you need details :)